agent/grpc/public/services/connectca/server_test.go

```diff
@@ -2,86 +2,42 @@ package connectca
 
 import (
 	"context"
-	"net"
-	"sync"
 	"testing"
-	"time"
 
 	"github.com/stretchr/testify/require"
 	"google.golang.org/grpc"
 
 	"github.com/hashicorp/consul/agent/consul/state"
 	"github.com/hashicorp/consul/agent/consul/stream"
-	"github.com/hashicorp/consul/agent/structs"
+	"github.com/hashicorp/consul/agent/grpc/public/testutils"
+	structs "github.com/hashicorp/consul/agent/structs"
 	"github.com/hashicorp/consul/proto-public/pbconnectca"
 )
 
 func noopForwardRPC(structs.RPCInfo, func(*grpc.ClientConn) error) (bool, error) {
 	return false, nil
 }
 
-func testStateStore(t *testing.T, publisher state.EventPublisher) *state.Store {
+func setupFSMAndPublisher(t *testing.T) (*testutils.FakeFSM, state.EventPublisher) {
```

Comment:
The bulk of what used to be here was moved to a shared location to reduce some of the boilerplate.